plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -216,30 +216,38 @@ def sweep(self, arguments: List[str]) -> None:
             for trial, ret in zip(trials, returns):
                 values: Optional[List[float]] = None
                 state: optuna.trial.TrialState = optuna.trial.TrialState.COMPLETE
-                func_error: Optional[Exception] = None
                 try:
                     if len(directions) == 1:
-                        values = [float(ret.return_value)]
+                        try:
+                            values = [float(ret.return_value)]
+                        except Exception:
```

Comment:
too broad. ValueError?

Reply:
Pro tip about re-raising exceptions (which is not critical here but good to know):

The following will preserve the current stack trace instead of starting a new one:
```
raise raise ValueError(...).with_traceback(sys.exc_info()[2])
```

This is not making a big deal here because the error is happening just two lines above the exception, but in many cases the error is caught far away from where it happens and rethrowing a new exception makes it hard to know what went wrong without careful debugging.

No need to apply in this case.

Reply:
> too broad. ValueError?

That must be fixed. Thanks. I replaced `Exception` with a pair of `ValueError` and `TypeError`.  I added `TypeError` because `float(None)` raises it in https://github.com/facebookresearch/hydra/pull/1294/commits/66b136d5378b20c42928fa69822c5d441fc96419.

> `raise raise ValueError(...).with_traceback(sys.exc_info()[2])`

Thank you for sharing the tips. I applied the suggestion in the commit https://github.com/facebookresearch/hydra/pull/1294/commits/66b136d5378b20c42928fa69822c5d441fc96419 because users can see the line of the error cause without opening the source code. If you find it verbose, I'll revert the change.

Previous
```console
  File "/prj/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py", line 251, in sweep
    raise e
  File "/prj/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py", line 226, in sweep
    f"Return value must be float-castable. Got '{ret.return_value}'."
ValueError: Return value must be float-castable. Got 'None'.
``` 

https://github.com/facebookresearch/hydra/pull/1294/commits/66b136d5378b20c42928fa69822c5d441fc96419
```console
  File "/prj/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py", line 227, in sweep
    ).with_traceback(sys.exc_info()[2])
  File "/prj/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py", line 223, in sweep
    values = [float(ret.return_value)]
ValueError: Return value must be float-castable. Got 'None'.
```